src/nrnpython/nrnpy_hoc.cpp

```diff
@@ -2703,8 +2703,8 @@ static PyObject* gui_helper_3_helper_(const char* name, Object* obj, int handle_
     for (int iarg = 0; iarg < narg; iarg++) {
         const int iiarg = iarg + 1;
         if (hoc_is_object_arg(iiarg)) {
-            PyObject* active_obj = nrnpy_ho2po(*hoc_objgetarg(iiarg));
-            PyTuple_SetItem(args.ptr(), iarg + 3, active_obj);
+            auto active_obj = nb::steal(nrnpy_ho2po(*hoc_objgetarg(iiarg)));
```

Comment:
Without any management of the ref counting, do you think it's worth using nb objects?

Reply:
Yes, it shows more clearly how the reference counting works. We own the reference returned by `nrnpy_ho2po`, then we pass our reference onto `PyTuple_SetItem`.

Esoteric reason: someone can now modify the code and place a sanity check (or any other code) that throws between the two lines.

Reply:
Last reason also occurred to me. 100% agree